src/vm/ceemain.cpp

```diff
@@ -1290,9 +1290,11 @@ void STDMETHODCALLTYPE EEShutDownHelper(BOOL fIsDllUnloading)
     }
 
 #ifdef FEATURE_PERFTRACING
-    // Shutdown the event pipe.
-    EventPipe::Shutdown();
-    DiagnosticServer::Shutdown();
+    if (!fIsDllUnloading)
```

Comment:
@brianrob would you know why `ETW::EnumerationLog::ProcessShutdown();` is called when `fIsDllUnloading` (the second time Shutdown is called). Isn't it dangerous to do cleanup there, am I missing something here? Could you please give us some background here on what's appropriate?